test/Tests/src/Data/Numbers_Spec.enso

```diff
@@ -181,28 +181,28 @@ spec =
             Integer.parse "-1234567" . should_equal -1234567
             Integer.parse "00000" . should_equal 0
             Integer.parse "00000123" . should_equal 123
-            Integer.parse "123.45" . should_fail_with Parse_Error_Data
-            Integer.parse "123A" . should_fail_with Parse_Error_Data
-            Integer.parse "aaaa" . should_fail_with Parse_Error_Data
+            Integer.parse "123.45" . should_fail_with Number_Parse_Error.Error
```

Comment:
```suggestion
            Integer.parse "123.45" . should_fail_with Number_Parse_Error
```

I know this is not allowed yet, but it would be nice if it were.

Wait actuall, can we rewrite `should_fail_with` to work with that? I think we already have the required tools (`Meta.type_of`).

Definitely not in scope of this PR, but thinking if we can schedule it (even with very low priority), as it's something we wanted to revisit at some point. Related stuff would be filtering errors in `Error.catch` and `Panic.catch` by the parent type and not by constructor too.

Reply:
Agree we should catch types not constructor (or possibly both).

Reply:
https://www.pivotaltracker.com/story/show/183839543